source/views/faqs/index.js

```diff
@@ -15,79 +14,60 @@ const styles = StyleSheet.create({
 	},
 })
 
-type Props = {}
+let fetchData = async (reload?: boolean) => {
+	let {text}: {text: string} = await fetch(API('/faqs'), {
+		delay: reload ? 500 : 0,
+	})
+		.json()
+		.catch(() => ({text: 'There was a problem loading the FAQs'}))
 
-type State = {
-	text: string,
-	loading: boolean,
-	refreshing: boolean,
+	return text
 }
 
-export class FaqView extends React.PureComponent<Props, State> {
-	static navigationOptions = {
-		title: 'FAQs',
-	}
-
-	state = {
-		text: '',
-		loading: true,
-		refreshing: false,
-	}
+export function FaqView() {
+	let [text, setText] = React.useState('')
+	let [loading, setLoading] = React.useState(true)
+	let [refreshing, setRefreshing] = React.useState(false)
 
-	componentDidMount() {
-		this.fetchData().then(() => {
-			this.setState(() => ({loading: false}))
+	React.useEffect(() => {
+		fetchData().then(text => {
+			setText(text)
+			setLoading(false)
 		})
-	}
+	}, [setText, setLoading])
+
+	let refresh = async (): any => {
+		setRefreshing(true)
 
-	fetchData = async (reload?: boolean) => {
-		let {text}: {text: string} = await fetch(API('/faqs'), {
-			forReload: reload ? 500 : 0,
+		await fetchData(true).then(text => {
+			setText(text)
 		})
-			.json()
-			.catch(() => ({text: 'There was a problem loading the FAQs'}))
 
-		this.setState(() => ({text}))
+		setRefreshing(false)
 	}
 
```

Comment:
can we put the 

```js
if (loading) {
	return <LoadingView />		
}
```

back here, instead of returning the result of the ternary, please?